server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1748,7 +1816,22 @@ private void updateWithLbRules(final DomainRouterJoinVO routerJoinVO, final Stri
                             .append(",protocol=").append(appLoadBalancerVO.getLbProtocol());
                 }
                 loadBalancingData.append(",stickiness=").append(getStickinessPolicies(firewallRuleVO.getId()));
-                loadBalancingData.append(",keepAliveEnabled=").append(offering.isKeepAliveEnabled()).append(",vmIps=");
+                if (isHttp != null) {
+                    loadBalancingData.append(",http=").append(isHttp);
+                } else if (firewallRuleVO.getSourcePortStart() == NetUtils.HTTP_PORT) {
+                    loadBalancingData.append(",http=").append(true);
+                }
+                if (isHttpKeepalive != null) {
+                    loadBalancingData.append(",keepAliveEnabled=").append(isHttpKeepalive);
+                } else {
+                    loadBalancingData.append(",keepAliveEnabled=").append(offering.isKeepAliveEnabled());
+                }
+                if (isTransparent != null) {
+                    loadBalancingData.append(",transparent=").append(isTransparent);
+                }
```

Comment:
This code seems repeated, but changing parameters; We could extract it to a method.